### PR TITLE
Basics of The Object Model

### PR DESCRIPTION
**Pull request by JudyNduati**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by JudyNduati** [comment]
This fixes #665

File: articles/basics-of-the-object-model/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- Multiple Inheritance
+- Hierarchical Inheritance
+- Multilevel Inheritance
+- Hybrid Inheritance.
+
+- [Polymorphism](https://www.tutorialspoint.com/java/java_polymorphism.htm#:~:text=Polymorphism%20is%20the%20ability%20of,to%20a%20child%20class%20object.&amp;text=In%20Java%2C%20all%20Java%20objects,and%20for%20the%20class%20Object.): The term polymorphism refers to the occurrence of something in many forms. Objects can be represented in many forms.
+
+For example,
+
+The world has many continents. Each continent consists of many countries, and each state is made of cities.
```
Comment (kayere) [inline articles/basics-of-the-object-model/index.md:58]: I don't really see how the example explains what polymorphism is. Consider using a different one.

File: articles/basics-of-the-object-model/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+Here are the significant elements of the Object Model.
+
+#### Abstraction
+Abstraction reduces complexity. It comes about from the recognition of similarities between objects. Abstraction takes place when the system stress details that are important to the user. It focuses mostly on the outside view of the object. Data is abstracted when protected by a set of methods, and only those methods can access data in the object.
+
+#### Encapsulation
+Encapsulation is achieved through information hiding or data hiding to reduce complexity and increase reusability. The user cannot see the inside of an object or a class, but the object can be accessed by calling the object's methods. Encapsulation and Abstraction are complementary concepts. In Abstraction, the system focus on object behavior and functionality. Encapsulation focus on implementation that gives rise to action.
+
+#### Hierarchy
+Different properties and functions form a class of the hierarchy. Through hierarchy, a class can be composed of inter-related sub-classes, which can have their sub-classes until the smallest level of components is reached. Upper-most classes more general, and classes at the base are more specific.
```
Comment (kayere) [inline articles/basics-of-the-object-model/index.md:97]: Could you please explain this concept further. It seems difficult to understand at a glance.

File: articles/basics-of-the-object-model/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+#### Modularity
+Modularity refers to dividing a program into components or modules to reduce the problem's complexity. Modularity takes place on broad and powerful applications with multiple classes. The modules help to manage complexity. Modularity focuses on implementation. Therefore, modularity and Encapsulation are related.
```
Comment (kayere) [inline articles/basics-of-the-object-model/index.md:100]: Please elaborate on how modularity and encapsulation are related.

File: articles/basics-of-the-object-model/index.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+### Conclusion
+That's all there is for this article. I hope this article gives you an understanding of the object model and its techniques.
```
Comment (kayere) [inline articles/basics-of-the-object-model/index.md:103]: Could you please make the conclusion a bit longer?

**Review by kayere — CHANGES_REQUESTED** [review]
Great article @JudyNduati, very informative. I have made a few suggestions. Please make the changes and the article will be good to go.

**Comment by hectorkambow** [comment]
👋 Hi @JudyNduati 
Thank you for submitting your "Basics of The Object Model" article - it looks like it will be a helpful article. 

Upon running your article through our 3rd party plagiarism checker, it raised a few red flags (more than we typically see) 
Please see the attachment. 

& be sure to consult our [plagiarism resource](https://www.scribbr.com/category/plagiarism/
[Basics of Object Model.pdf](https://github.com/section-io/engineering-education/files/5619488/Basics.of.Object.Model.pdf)
) to get a clear understanding on how to avoid it.

**Comment by JudyNduati** [comment]
Thank you @kayere and @hectorkambow, I made the changes you suggested and worked on the red flags. Kindly have a review.

**Comment by JudyNduati** [comment]
Thank you @kayere and @hectorkambow, I made the changes you suggested and worked on the red flags. Kindly have a review

**Review by kayere — APPROVED** [review]
Great work making the changes @JudyNduati.

**Pull request merged**